Python/bytecodes.c

```diff
@@ -790,10 +791,12 @@ dummy_func(
             _PyInterpreterFrame *dying = frame;
             frame = tstate->current_frame = dying->previous;
             _PyEval_FrameClearAndPop(tstate, dying);
-            frame->prev_instr += frame->return_offset;
+            frame->instr_ptr += frame->return_offset;
             _PyFrame_StackPush(frame, retval);
             LOAD_SP();
             LOAD_IP();
+            frame->instr_ptr -= frame->return_offset;
+            frame->return_offset = 0;
```

Comment:
Here we need to update instr_ptr before LOAD_IP, so that next_instr gets set with the right value, and then revert the change in instr_ptr so that line tracing works. If we stick with the current scheme we should change the macros so that this dance is no longer necessary. (basically add an incrementing version of LOAD_IP).

Reply:
Agreed -- this dance looks confusing.